### PR TITLE
Add simple cases of as_strided and as_strided_ to ATen XLA tensor

### PR DESCRIPTION
**Pull request by asuhan**

### CODE REVIEW

File: torch_xla/csrc/ops/as_strided.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+#include "tensorflow/compiler/xla/types.h"
+#include "torch_xla/csrc/ir.h"
+
+namespace torch_xla {
+namespace ir {
+namespace ops {
+
+class AsStrided : public Node {
+ public:
+  AsStrided(const Value& input, tensorflow::gtl::ArraySlice<xla::int64> size,
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/as_strided.h]: Pass in vector<> and move.

File: torch_xla/csrc/tensor.h [hunk 2]
```diff
@@ -267,6 +267,18 @@ class XLATensor {
   static XLATensor narrow(const XLATensor& input, xla::int64 dim,
                           xla::int64 start, xla::int64 length);
 
+  // Takes a slice from the input as R1 at the specified offset and reshapes it
+  // into the provided size.
+  static XLATensor as_strided(
+      const XLATensor& input,
+      tensorflow::gtl::ArraySlice<const xla::int64> size,
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor.h]: There is written nowhere this has to be array slices.
These APIs has only one call site, which is already creating a move-able vector.
I'd just pass that through down to the op.

File: torch_xla/csrc/ops/as_strided.cpp [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+namespace ops {
+namespace {
+
+xla::XlaOp LowerAsStrided(const xla::XlaOp& input,
+                          tensorflow::gtl::ArraySlice<xla::int64> size,
+                          c10::optional<xla::int64> storage_offset) {
+  xla::int64 input_element_count =
+      xla::ShapeUtil::ElementsIn(XlaHelpers::ShapeOfXlaOp(input));
+  xla::int64 slice_size = std::accumulate(
+      size.begin(), size.end(), xla::int64(1), std::multiplies<xla::int64>());
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/as_strided.cpp]: xla::util::Multiply<>

File: torch_xla/csrc/ops/as_strided.cpp [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    tensorflow::gtl::ArraySlice<xla::int64> stride) {
+  XLA_CHECK_EQ(size.size(), stride.size());
+  std::vector<xla::int64> expected_stride;
+  expected_stride.reserve(stride.size());
+  expected_stride.push_back(1);
+  std::partial_sum(size.rbegin(), size.rend() - 1,
+                   std::back_inserter(expected_stride),
+                   std::multiplies<xla::int64>());
+  std::reverse(expected_stride.begin(), expected_stride.end());
+  return xla::util::ToVector<xla::int64>(stride) == expected_stride;
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/as_strided.cpp]: You should be able to compare stride directly.

File: torch_xla/csrc/ops/as_strided.cpp [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    tensorflow::gtl::ArraySlice<xla::int64> size,
+    tensorflow::gtl::ArraySlice<xla::int64> stride) {
+  XLA_CHECK_EQ(size.size(), stride.size());
+  std::vector<xla::int64> expected_stride;
+  expected_stride.reserve(stride.size());
+  expected_stride.push_back(1);
+  std::partial_sum(size.rbegin(), size.rend() - 1,
+                   std::back_inserter(expected_stride),
+                   std::multiplies<xla::int64>());
+  std::reverse(expected_stride.begin(), expected_stride.end());
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/as_strided.cpp]: This is a bit simpler:
```
std::vector<xla::int64> expected_stride(size.size(), 1);                                                                
for (size_t i = size.size() - 1; i > 0; --i) {                                                                          
  expected_stride[i - 1] = expected_stride[i] * size[i];                                                                
} 
```

File: torch_xla/csrc/ops/as_strided.h (hunk 1 above)
Comment (asuhan) [reply torch_xla/csrc/ops/as_strided.h]: Done.

File: torch_xla/csrc/tensor.h (hunk 2 above)
Comment (asuhan) [reply torch_xla/csrc/tensor.h]: Done.

File: torch_xla/csrc/ops/as_strided.cpp (hunk 3 above)
Comment (asuhan) [reply torch_xla/csrc/ops/as_strided.cpp]: Done.

File: torch_xla/csrc/ops/as_strided.cpp (hunk 4 above)
Comment (asuhan) [reply torch_xla/csrc/ops/as_strided.cpp]: Done.

File: torch_xla/csrc/ops/as_strided.cpp (hunk 5 above)
Comment (asuhan) [reply torch_xla/csrc/ops/as_strided.cpp]: Done.

**Pull request merged**